### PR TITLE
analytics(weave): copy code event, colab event

### PR DESCRIPTION
**Pull request by jwlee64**
https://wandb.atlassian.net/browse/WB-19281

swizzled code block code (to get to the copy button). This is mechanism provided by docusaurus, that copies an internal component and allows us to modify it.
Added colab event
Added an event to the swizzled copy button

### CODE REVIEW

File: docs/src/theme/CodeBlock/CopyButton/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import IconSuccess from '@theme/Icon/Success';
+import styles from './styles.module.css';
+export default function CopyButton({code, className}) {
+  const [isCopied, setIsCopied] = useState(false);
+  const copyTimeout = useRef(undefined);
+  const handleCopyCode = useCallback(() => {
+
+    // Track the copy event
+    // Only line added to the swizzled component
+    window.analytics?.track('copy_code', {code: typeof code == 'string' ? code.slice(0, 100) : 'code is not string'});
```
Comment (jwlee64) [inline docs/src/theme/CodeBlock/CopyButton/index.js:15]: this is the second line added

File: docs/docs/introduction.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ Our goal is to bring rigor, best-practices, and composability to the inherently
 
 ![Weave Hero](../static/img/weave-hero.png)
 
-Seriously, try the 🍪 **[quickstart](/quickstart)** 🍪 or <a class="vertical-align-colab-button" target="_blank" href="http://wandb.me/weave_colab"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
+Seriously, try the 🍪 **[quickstart](/quickstart)** 🍪 or <a class="vertical-align-colab-button" target="_blank" href="http://wandb.me/weave_colab" onClick={()=>{window.analytics?.track("Weave colab clicked")}}><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>
```
Comment (jwlee64) [inline docs/docs/introduction.md:18]: this is the first line changed

File: docs/src/theme/CodeBlock/CopyButton/index.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import copy from 'copy-text-to-clipboard';
+import {translate} from '@docusaurus/Translate';
+import IconCopy from '@theme/Icon/Copy';
+import IconSuccess from '@theme/Icon/Success';
+import styles from './styles.module.css';
+export default function CopyButton({code, className}) {
+  const [isCopied, setIsCopied] = useState(false);
+  const copyTimeout = useRef(undefined);
+  const handleCopyCode = useCallback(() => {
+    // Track the copy event
```
Comment (jwlee64) [inline docs/src/theme/CodeBlock/CopyButton/index.js:12]: second lines changed

**Review by tssweeney — APPROVED** [review]
Approving, but consider making a more standard format for event names. Maybe something like `SOURCE: TARGET - ACTION` or something so we can stay organized downstream.

**Pull request merged**